Mathlib/Topology/Compactification/OnePointEquiv.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2024 Bjørn Kjos-Hanssen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bjørn Kjos-Hanssen
+-/
+import Mathlib.LinearAlgebra.Projectivization.Basic
+import Mathlib.Topology.Compactification.OnePoint
+
+/-!
+# One-point compactification and projectivization
+
+We construct a set-theoretic equivalence between
+`OnePoint K` and the projectivization `ℙ K (Fin 2 → K)` for an arbitrary field `K`.
+
+TODO: Add the extension of this equivalence to a homeomorphism in the case `K = ℝ`,
+where `OnePoint ℝ` gets the topology of one-point compactification.
+
+
+## Main definitions
+
+* `divOnePoint`: Division on `K` taking values in `OnePoint K`. Notation: `÷`.
+* `divSlope`: Lowering `divOnePoint` to a map `ℙ K (Fin 2 → K) → OnePoint K`.
+
+## Main results
+
+* `divSlope_equiv` proves that `ℙ K (Fin 2 → K) ≃ OnePoint K`.
+
+## Tags
+
+one-point extension, projectivization
+-/
+
+open scoped LinearAlgebra.Projectivization OnePoint
+open Projectivization Classical
+
+/-!
+### Definition and basic properties
+
+We define a division operation `divOnePoint`, with notation `÷`.
+It takes inputs from a field `K` and outputs a value in `K ∪ {∞}`, i.e.,  `OnePoint K`.
+We prove that `÷` preserves projective equivalence: `(c•a)÷(c•b)=a÷b`.
+Hence `÷` projects down to a map `divSlope` from `ℙ K (Fin 2 → K)` to `OnePoint K`
+that takes a line through the origin and returns its slope. For a vertical line
+the slope is `∞`.
+-/
+variable {K : Type*}
+/-- A modified division from a structure with `Zero` and `Div` to its
+ `OnePoint` extension. With notation `÷` for `divOnePoint`,
+  we have in particular that `1 ÷ 0 = ∞`. -/
+noncomputable def divOnePoint [Zero K] [Div K] (a : K) (r : K): OnePoint K :=
+  if r ≠ 0 then a / r else ∞
```

Comment:
If we want this setup to work in the non-commutative setting (i.e., for `[DivisionRing K]` instead of `[Field K]`) then this should be:
```suggestion
noncomputable def divOnePoint [Zero K] [Inv K] [Mul K] (a : K) (r : K) : OnePoint K :=
  if r ≠ 0 then r⁻¹ * a else ∞
```

Reply:
Does `divSlope_inj_lifted` hold for division rings? My proof doesn't seem to go through in that generality since it reduces to '(a.1 1)⁻¹ * a.1 0 = (b.1 1)⁻¹ * b.1 0` implying `(b.1 1)⁻¹ * a.1 1 * b.1 0 = a.1 0`.